Reference/Common-Pitfalls/index.md

```diff
@@ -0,0 +1,466 @@
+
+How do you know if you are using Dynamics?
+
+* If you are using `@CurrentPage` then __you are__ using dynamics
+* If you are using the UmbracoHelper query methods like `@Umbraco.Content` or `@Umbraco.Media` instead of the typed methods like `@Umbraco.TypedContent` and `@Umbraco.TypedMedia` then __you are__ using dynamics
+
+It is strongly advised that you use the strongly typed `@Model.Content` instead of `@CurrentPage` models in your views,  
+this will actually perform much better and you'll be forward compatible with Umbraco v8+ with regards to querying `IPublishedContent`. 
+
+A large problem with the performance of dynamics is having to parse string syntax such as:
+`@CurrentPage.Children.Where("DocumentTypeAlias == \"DatatypesFolder\" && Visible")` and turn that into something that is compilable when 
+instead it can just be written as something that compiles
+
+## Using the Services layer in your views
+
+The Services layer of Umbraco is for manipulating the business logic of Umbraco directly to/from the database. 
+None of these methods should be used within your views and can have a very large impact on performance and stability of 
+your application.
+
+Your views should rely only on the readonly data access of the `UmbracoHelper` and the properties/methods that it exposes. This ensures
+that the data being queried is fast (comes from cache) and that you aren't inadvertently making database changes.
+
+If you are using `Application.Services...` in your views, you should figure out why this is being done and in most cases remove this logic.   
+
+## Using UmbracoContext to access ApplicationContext
+
+You should not access the `ApplicationContext` via the `UmbracoContext`. 
+
+For example: `UmbracoContext.Current.Application` _<-- this is now deprecated/obsolete_
+
+If you need access to both the `UmbracoContext` and the `ApplicationContext` you should either 
+
+* Access these services via the properties exposed on the Umbraco base class you are using (i.e. Controllers, views, controls, http handler, etc...)
+* or inject these services into the services you are using 
+* or access each of these services from their own singleton constructs: `UmbracoContext.Current` and `ApplicationContext.Current`.
+
+## Using Umbraco content items for volatile data 
+
+This is one of the worst Umbraco anti-patterns and could very well cause your site to perform ultra poorly. 
+
+Umbraco's content should not be used for volatile data, Umbraco's APIs and the way Umbraco's data is persisted
+was never designed for this. If you need to store/write/track data that changes a lot you should use a 
+custom database table or another service but not Umbraco content nodes.
+
+Some examples of what not to do are:
+
+* Hit counters to track the number of times your page has been viewed - use something like Google Analytics for this or a custom database table
```

Comment:
Good examples here of what not to do ;)

Luckily we don't automatically turn on the Umbraco Forms workflow for saving responses as child nodes and you have to explicitly opt into this & create a workflow to create nodes for you.